x/dualstaking/keeper/delegator_reward.go

```diff
@@ -244,43 +249,44 @@ func (k Keeper) rewardDelegator(ctx sdk.Context, delegation types.Delegation, am
 		delegatorReward.Provider = delegation.Provider
 		delegatorReward.Delegator = delegation.Delegator
 		delegatorReward.ChainId = delegation.ChainID
-		delegatorReward.Amount = sdk.NewCoin(k.stakingKeeper.BondDenom(ctx), amount)
+		delegatorReward.Amount = amount
```

Comment:
there can be different types of coins, you need to change the proto of delegatorReward accordingly (from coin to coins) and make a migrator